test/extended/util/framework.go

```diff
@@ -2075,3 +2083,53 @@ func DoesApiResourceExist(config *rest.Config, apiResourceName, groupVersionName
 func groupName(groupVersionName string) string {
 	return strings.Split(groupVersionName, "/")[0]
 }
+
+func collectConfigManifestsFromDir(configManifestsDir string) (error, []runtime.Object) {
+	objects := []runtime.Object{}
+	files, err := ioutil.ReadDir(configManifestsDir)
+	if err != nil {
+		return err, nil
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			klog.V(2).Infof("%q is not a file, skipping", file.Name())
+			continue
+		}
+
+		body, err := ioutil.ReadFile(configManifestsDir + "/" + file.Name())
+		if err != nil {
+			return err, nil
+		}
+
+		object := &metav1.TypeMeta{}
+		err = yaml.Unmarshal(body, &object)
+		if err != nil {
+			return err, nil
+		}
+
+		if object.APIVersion == "config.openshift.io/v1" {
+			switch object.Kind {
+			case "Infrastructure":
```

Comment:
need to fail if you find more than one of a given group,version,kind,namespace,name

Reply:
Done